src/Microsoft.AspNetCore.ResponseCaching/Internal/ResponseCachingKeyProvider.cs

```diff
@@ -124,6 +126,10 @@ public string CreateStorageVaryByKey(ResponseCachingContext context)
 
                         for (var j = 0; j < headerValues.Count; j++)
                         {
+                            if (j > 0)
```

Comment:
Per RFC:

>A recipient MAY combine multiple header fields with the same field
   name into one "field-name: field-value" pair, without changing the
   semantics of the message, by appending each subsequent field value to
   the combined field value in order, separated by a comma.

So the values of the headers will not be delimited.